src/dos/dos_locale.cpp

```diff
@@ -1404,9 +1404,12 @@ static void load_keyboard_layout()
 
 	if (using_detected) {
 		const auto& host_locale = GetHostKeyboardLayouts();
-		assert(!host_locale.log_info.empty());
-		LOG_MSG("LOCALE: Keyboard layout and code page detected from '%s'",
-		        host_locale.log_info.c_str());
+		if (host_locale.log_info.empty()) {
```

Comment:
I would prefer to keep the assert - we want the host-specific code to fill-in the `log_info`, so that the log contains the information telling why the selected keyboard layout was even considered in the first place.

We should probably move the `LOG_MSG` to more appropriate place - please wait for me with this, I should have time to look into this today evening, or latest tomorrow.

Reply:
I have prepared the fix preserving the assertion - https://github.com/dosbox-staging/dosbox-staging/tree/fc/kb-autodetect-fixes-1 - just need some more testing.

I also came to conclusion that preferring CPI files font over ROM font for auto-detected keyboard layouts was probably not a good idea. I have changed this, too.